census/tests/test_census.py

```diff
@@ -121,6 +121,67 @@ def test_la_canada_2015(self):
         )
 
 
+class TestCodedValues(CensusTestCase):
+    """
+    Tests for handling coded values, like -666666666 and -999999999.
+    """
+    def test_handle_666666666(self):
+        """
+        Test the default behavior of handling -666666666 values, which is to
+        cast them to null.
+        """
+        # This call should return a value of -666666666
+        return_val = self._client.acs5.state_county_tract('B19081_001E',
+                                                          42,
+                                                          101,
+                                                          '989100',
+                                                          year=2016)
+        self.assertEqual(return_val[0]['B19081_001E'], None)
+
+    def test_handle_666666666_as_error(self):
+        """
+        Test raising an error for -666666666 values.
+        """
+        with self.assertRaises(CensusException):
+            return_val = self._client.acs5.state_county_tract('B19081_001E',
+                                                              42,
+                                                              101,
+                                                              '989100',
+                                                              year=2016,
+                                                              cast_nulls=False)
+
+    def test_handle_666666666_as_null(self):
+        """
+        Test casting -666666666 values to null.
+        """
+        return_val = self._client.acs5.state_county_tract('B19081_001E',
```

Comment:
We should also test that a warning is emitted that a value was cast to a null. Particularly if we decide to make this the default behavior.